v2/client/client.go

```diff
@@ -233,9 +233,14 @@ func (c *ceClient) StartReceiver(ctx context.Context, fn interface{}) error {
 					continue
 				}
 
-				if err := c.invoker.Invoke(ctx, msg, respFn); err != nil {
-					cecontext.LoggerFrom(ctx).Warnf("Error while handling a message: %s", err)
-				}
+				// Do not block on the invoker.
+				wg.Add(1)
```

Comment:
Do we really need to synchronize the `Invoke` on the wg? If `Invoke` ends after these goroutines loop, is this a potential issue?

Reply:
it just lets us know to block if invoke is happening when the process is spinning down

Reply:
that's my point, do we need to wait for invoke to finish before spinning down these goroutines? isn't this done automatically by the runtime?

Reply:
It is not done automatically. We would return from this loop and exit the process before the invokers had a chance to finish on a soft sig-term